spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -135,43 +163,74 @@ public TwoStepsConversions(CustomConversions customConversions, ObjectToKeyFacto
 				throw new DatastoreDataException("Unable process elements of a collection", e);
 			}
 		}
+		return (T) readConverter.apply(val, targetComponentType);
+	}
 
-		return readConverter.apply(val, targetCollectionType == null ? targetComponentType : targetCollectionType);
+	private <T, R> Map<T, R> convertOnReadSingleEmbeddedMap(Object value,
+			Class<T> keyType, TypeInformation<R> targetComponentType) {
+		Assert.notNull(value, "Cannot convert a null value.");
+		if (value instanceof BaseEntity) {
+			return this.datastoreEntityConverter.readAsMap(keyType, targetComponentType,
+					(BaseEntity) value);
+		}
+		throw new DatastoreDataException(
+				"Embedded entity was expected, but " + value.getClass() + " found");
 	}
 
 	@SuppressWarnings("unchecked")
-	private <T> T convertOnReadSingleEmbedded(Object value, Class<?> targetClass) {
+	private <T> T convertOnReadSingleEmbedded(Object value,
```

Comment:
no need to split into 2 lines

Reply:
For me it looks like it's just 1 character too long if I don't